Sources/Subprocess/API.swift

```diff
@@ -142,6 +142,74 @@ public func run<
 
 // MARK: - Custom Execution Body
 
+/// Run an executable with given parameters and a custom closure
+/// to manage the running subprocess' lifetime and its IOs.
+/// - Parameters:
+///   - executable: The executable to run.
+///   - arguments: The arguments to pass to the executable.
+///   - environment: The environment in which to run the executable.
+///   - workingDirectory: The working directory in which to run the executable.
+///   - platformOptions: The platform specific options to use
+///     when running the executable.
+///   - input: The input to send to the executable.
+///   - output: How to manage the executable standard ouput.
+///   - error: How to manager executable standard error.
+///   - isolation: the isolation context to run the body closure.
+///   - body: The custom execution body to manually control the running process
+/// - Returns a ExecutableResult type containing the return value
+///     of the closure.
+#if SubprocessSpan
+@available(SubprocessSpan, *)
+#endif
+public func run<Result, Input: InputProtocol, Output: OutputProtocol, Error: OutputProtocol>(
+    _ executable: Executable,
+    arguments: Arguments = [],
+    environment: Environment = .inherit,
+    workingDirectory: FilePath? = nil,
+    platformOptions: PlatformOptions = PlatformOptions(),
+    input: Input = .none,
+    output: Output,
+    error: Error,
+    isolation: isolated (any Actor)? = #isolation,
+    body: ((Execution, AsyncBufferSequence, AsyncBufferSequence) async throws -> Result)
```

Comment:
Since you are accepting output and error explicitly, the body closure shouldn't have AsyncBufferSequence` because that's fundamentally incompatible with explicit output types. Here `body` should be just `(Execution) async throws -> Result`

Reply:
Done!

Reply:
However, how do you reconcile that with the fact that other overloads DO include the AsyncBufferSequence in the body closure? Should those be removed, then?